File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        //Create vector and go through prefixes
        vector<int> result(nums.size());
        int block = 1; 
        result[0] = 1;
        //Prefixes
        for(int i = 1; i < nums.size(); i++) {
            block *= nums[i - 1];
            result[i] = block;
        }
        block = 1;
        //Postfixes
        for(int j = nums.size() - 2; j >= 0; j--) {
            block *= nums[j + 1];
            result[j] *= block;
        }

        return result;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        result = [0] * len(nums)
        result[0] = nums[0]
        for i in range(1, len(nums)):
            result[i] = max(result[i -1] + nums[i], nums[i])
            print(result[i])

        return result[-1]